backend/organization/models/content.py

```diff
@@ -0,0 +1,137 @@
+from django.db import models
+from organization.models import Project
+from django.contrib.auth.models import User
+
+
+class Posts(models.Model):
+    project = models.ForeignKey(
+        Project,
+        related_name="post_project",
+        verbose_name="Project",
+        help_text="Points to post of the project",
+        on_delete=models.PROTECT
+    )
+
+    author_user = models.ForeignKey(
+        User,
+        related_name="post_author",
+        help_text='Points to a user who wrote the post',
+        verbose_name="Author",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True
+    )
+
+    content = models.TextField(
+        help_text="Content of the post",
+        verbose_name="Content"
+    )
+
+    created_at = models.DateTimeField(
+        help_text="Time when post was created",
+        verbose_name="Created at",
+        auto_now_add=True
+    )
+
+    updated_at = models.DateTimeField(
+        help_text="Time when post was updated",
+        verbose_name="Updated at",
+        auto_now=True
+    )
+
+    is_hidden = models.BooleanField(
+        help_text="If post has made hidden from public",
+        verbose_name="Is hidden?",
+        default=False
+    )
+
+    deleted_at = models.DateTimeField(
+        help_text="Time when post was deleted",
+        verbose_name="Deleted at",
+        null=True,
+        blank=True
+    )
+
+    deleted_by_user = models.ForeignKey(
+        User,
+        help_text="User who deleted the post",
+        verbose_name="Deleted by user",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
```

Comment:
The `on_delete` parameter here is when user is deleted from the database, we still want to preserve posts. So it will set user's foreign key to be null.

Reply:
https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.ForeignKey.on_delete

Reply:
Got it, makes sense!